docs/book/v1/intro.md

```diff
@@ -23,17 +23,30 @@ interface UserInterface
 {
     /**
      * Get the unique user identity (id, username, email address or ...)
-     *
-     * @return string
      */
-    public function getIdentity(): string;
+    public function getIdentity() : string;
+
+    /**
+     * Set the user identity
+     */
+    public function setIdentity(string $identity) : void
```

Comment:
I'm a little skeptical of adding setter methods to the `UserInterface`. The reason is because this could potentially lead to cases where middleware resets the identify, roles, etc. — which is fine for altering the user instance passed _deeper_ into the request layer, but could have ramifications for _parent_ layers as well. I think we should prefer immutability here.